diff --git a/Makefile b/Makefile
index 6194fe3..6f0f661 100644
--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,7 @@ CC_CUDA110 += -gencode arch=compute_80,code=sm_80
 CC_CUDA11x := -gencode arch=compute_75,code=sm_75
 CC_CUDA11x += -gencode arch=compute_80,code=sm_80
 CC_CUDA11x += -gencode arch=compute_86,code=sm_86
+CC_CUDA11x += -gencode arch=compute_89,code=sm_89

 CC_cublasLt110 := -gencode arch=compute_75,code=sm_75
 CC_cublasLt110 += -gencode arch=compute_80,code=sm_80
@@ -55,6 +56,7 @@ CC_cublasLt110 += -gencode arch=compute_80,code=sm_80
 CC_cublasLt111 := -gencode arch=compute_75,code=sm_75
 CC_cublasLt111 += -gencode arch=compute_80,code=sm_80
 CC_cublasLt111 += -gencode arch=compute_86,code=sm_86
+CC_cublasLt111 += -gencode arch=compute_89,code=sm_89


 all: $(ROOT_DIR)/dependencies/cub $(BUILD_DIR) env
diff --git a/bitsandbytes/cextension.py b/bitsandbytes/cextension.py
index 264e899..e42d8b8 100644
--- a/bitsandbytes/cextension.py
+++ b/bitsandbytes/cextension.py
@@ -41,7 +41,7 @@ class CUDASetup(object):
         elif self.cuda_version_string[:2] == '11' and int(self.cuda_version_string[2]) > 0:
             make_cmd += ' make cuda11x'

-        has_cublaslt = self.cc in ["7.5", "8.0", "8.6"]
+        has_cublaslt = self.cc in ["7.5", "8.0", "8.6", "8.9"]
         if not has_cublaslt:
             make_cmd += '_nomatmul'

diff --git a/bitsandbytes/cuda_setup/main.py b/bitsandbytes/cuda_setup/main.py
index 6a6bc74..7258f16 100644
--- a/bitsandbytes/cuda_setup/main.py
+++ b/bitsandbytes/cuda_setup/main.py
@@ -143,7 +143,7 @@ def evaluate_cuda_setup():
         cuda_setup.add_log_entry(f'CUDA SETUP: Detected CUDA version {cuda_version_string}')

     # 7.5 is the minimum CC vor cublaslt
-    has_cublaslt = cc in ["7.5", "8.0", "8.6"]
+    has_cublaslt = cc in ["7.5", "8.0", "8.6", "8.9"]

     # TODO:
     # (1) CUDA missing cases (no CUDA installed by CUDA driver (nvidia-smi accessible)
